Cleanup readme and docs for using docs.dat-data.com

joehand opened the pull request:
Were updating this readme to be the landing page of docs.dat-data.com and removing the read the docs site. [See the rendered readme](https://github.com/joehand/dat/tree/cleanup-docs#dat)

This PR:
* removes read the docs files
* consolidate docs to readme
* update readme to be homepage for docs.dat-data.com
* adds quick demo + gif

karissa commented:
Awesome! Some tiny things I found below. I'd commit directly but its on your branch 

* 'awesome dat' might be better renamed to 'ecosystem'.. not entirely clear what that means to the uninitiated :)
*  https://docs.dat-data.com doesn't work (getting https error) maybe we should just make it http:// and it'll upgrade automagically to https when/if we get that working
* Let's not link to 'dat.land' quite yet. 
* Maybe the table of contents doesn't need to be collapsed because its so short
* Maybe `npm install -g dat` should be higher up ('above the fold') somehow, with a link to the troubleshooting section. The output of the `dat` cmd could be much later, i think after the 'getting started section', because the good part of the 'getting started' section is you walk through each command at a time and it might be easier to digest.

joehand commented:
Thanks! Changed those things. Do you think we should remove the "ecosystem" section too? It feels a little lonely right now.

karissa commented:
Yeah I agree. I think we have three ecosystem sections now... the 'awesome dat' repo, 'ecosystem' section in the docs, and this section.. maybe we should only have one :) and then link to it. 

Maybe we could have minidocs use the awesome-dat repo as the ecosystem section? And then link to it.

Also I just noticed that having the table of contents reference the 'getting started' section doesnt make as much sense cause its at the top.  Perhaps the table of contents should be before the getting started section

joehand commented:
Removed ecosystem and moved TOC up. 

I added a Dat CLI title so hopefully it'll be clean on docs.dat-data.com when things shift from being dat generally to CLI specific. The general intro to Dat could be strengthened a bit for the docs site but I think its okay for now.

mafintosh commented on README.md:
```diff
@@ -84,17 +153,95 @@ The Share Link is secret and only those you share it with will be able to get th
+
+The `-g` option installs Dat globally allowing you to run it as a command. Make sure you installed with that option.
+
+* If you receive an `EACCES` error, read [this guide](https://docs.npmjs.com/getting-started/fixing-npm-permissions) on fixing npm permissions.
```
rerun it with sudo?
